doc/vxlan/EVPN/EVPN_VxLAN_Multihoming.md

```diff
@@ -0,0 +1,1639 @@
+
+The following are targeted scale numbers for EVPN VxLAN Multihoming on platforms providing EVPN Multihoming capability:
+1. Support up to 1K EVPN ESIs in a given site/cluster.
+2. Support up to 128 EVPN ESIs on a given VTEP.
+3. Support up to 4 peering VTEP per Ethernet-Segment.
+
+<a id="115-Convergence-Requirements"></a>
+### 1.1.5 Convergence Requirements
+
+Achieve sub-second convergence in following failure scenarios:
+1. Link failure
+2. Port failure
+3. Core isolation
+4. VTEP restart
+
+<a id="12-Design-Overview"></a>
+## 1.2 Design Overview
+EVPN Multihoming feature provides [RFC 7432](https://datatracker.ietf.org/doc/html/rfc7432) and [RFC 8365](https://datatracker.ietf.org/doc/html/rfc8365) based solution for supporting all-active redundancy in EVPN VxLAN network. The all-active multihomed device(s) are required to be connected to the EVPN VxLAN network with Link-aggregation (LAG) running LACP protocol. Subsequent sub-sections below cover typical traffic forwarding cases.
+
+<a id="121-Basic-Approach"></a>
+### 1.2.1 Basic Approach
+The diagram below shows a typical EVPN Multihoming network with Multihomed (MH)  and Single-homed (SH) devices connected to EVPN VxLAN VTEPs.
+
+![](images/evpn_mh_diagram.PNG)
+
+__Figure 1: Typical EVPN Multihoming network__
+
+The devices H1, H4, and H5 are SH devices and connected to Vtep-1, Vtep-4, and Vtep-5 respectively. The devices H2 & H3 are multihomed and connected to EVPN VxLAN network with active-active redundancy using LAG. Device H2 is multihomed to Vtep-1 and Vtep-4 whereas device H3 is multihomed to Vtep-1, Vtep-2, Vtep-3, and Vtep-4.
+
+In order to configure EVPN Multihoming for the LAGs connecting multihomed devices, Type-0, Type-1, or Type-3 ESI is required to be configured on the LAG interfaces:
+- *T*ype-0: Operator configured ESI*
+- *Type-1: ESI automatically derived from LACP partner MAC*
+- *Type-3: ESI automatically derived from configured LAG system-mac address*
+
+For a given Multihomed Ethernet-Segment, the value of ES-ID should be same on all of the participating VTEPs.
+
+For simplicity, assume Type-3 ESI is configured on the PortChannel interfaces shown in the above diagram. The Vtep-1 and Vtep-4 LAG configuration will look like this:
+
+```
+sudo config interface PortChannel1 sys-mac 00:00:00:00:11:11
+sudo config interface PortChannel1 evpn-esi auto-system-mac
+
+sudo config interface PortChannel2 system-mac 00:00:00:00:22:22
+sudo config interface PortChannel2 evpn-esi auto-system-mac
+```
+
+Whereas, on Vtep-2 and Vtep-3 only PortChannel2 config is required since only PortChannel2 is multihomed to these VTEPs.
+```
+sudo config interface PortChannel2 system-mac 00:00:00:00:22:22
+sudo config interface PortChannel2 evpn-esi auto-system-mac
+```
+The Type-3 ESI value will be derived by combining the configured system-mac address on the PortChannel interface, and the PortChannel number. The system-mac is required to be same on all of the VTEPs multihoming a given LAG.
```

Comment:
And also the PortChannel number has to be same on all of the VTEPs multihoming right?

Reply:
If you are using ESI Type-3, the portchannel Id is used a local discriminator. However, ESI type-0 can be use where the whole 10 bytes are configurable. The portChannel id is no longer relevant.